packages/devextreme/js/ui/tab_panel.js

```diff
@@ -252,6 +273,51 @@ const TabPanel = MultiView.inherit({
         this._focusTarget().attr('tabIndex', -1);
     },
 
+    _getTabsOrientation() {
+        const { tabsPosition } = this.option();
+
+        if([TABS_POSITION.top, TABS_POSITION.bottom].includes(tabsPosition)) {
+            return TABS_ORIENTATION.horizontal;
+        }
+
+        if([TABS_POSITION.right, TABS_POSITION.left].includes(tabsPosition)) {
+            return TABS_ORIENTATION.vertical;
+        }
+    },
+
+    _getTabPanelTabsPositionClass(value = null) {
+        value = value ?? this.option('tabsPosition');
+
+        switch(value) {
+            case TABS_POSITION.right:
+                return TABPANEL_TABS_POSITION_CLASS.right;
+            case TABS_POSITION.bottom:
+                return TABPANEL_TABS_POSITION_CLASS.bottom;
+            case TABS_POSITION.left:
+                return TABPANEL_TABS_POSITION_CLASS.left;
+            case TABS_POSITION.top:
+            default:
+                return TABPANEL_TABS_POSITION_CLASS.top;
+        }
+    },
+
+    _toggleTabPanelTabsPositionClass(prevPosition) {
+        const prevClass = this._getTabPanelTabsPositionClass(prevPosition);
+        const newClass = this._getTabPanelTabsPositionClass();
+
+        if(prevClass) {
```

Comment:
Whether it was or wasn't doesn't matter, we just need to clear the class

```
for (const position in TABS_POSITION) {
    this.$element().removeClass(this._getTabPanelTabsPositionClass(position));
}
```

Reply:
We shouldn't remove all TABS_POSITION classes in for cycle. Just a one class.

Reply:
[feccc1664117de61365ba2df824d600fa6ed06f6]